src/state/internal/createQueryStore.ts

```diff
@@ -0,0 +1,778 @@
+   * Forces a fetch request even if the current data is fresh and not stale.
+   * If `true`, the fetch operation bypasses existing cached data.
+   */
+  force?: boolean;
+  /**
+   * Overrides the default stale duration for this fetch, in milliseconds.
+   * If the fetch is successful, the subsequently scheduled refetch will occur after
+   * the specified duration.
+   */
+  staleTime?: number;
+}
+
+/**
+ * Represents an entry in the query cache, which stores fetched data along with metadata, and error information
+ * in the event the most recent fetch failed.
+ */
+interface CacheEntry<TData> {
+  data: TData | null;
+  errorInfo: {
+    error: Error;
+    lastFailedAt: number;
+    retryCount: number;
+  } | null;
+  lastFetchedAt: number;
+}
+
+/**
+ * A specialized store interface that combines Zustand's store capabilities with remote data fetching support.
+ *
+ * In addition to Zustand's store API (such as `getState()` and `subscribe()`), this interface provides:
+ * - **`enabled`**: A boolean indicating if the store is actively fetching data.
+ * - **`fetch(params, options)`**: Initiates a data fetch operation.
+ * - **`getData(params)`**: Returns the cached data, if available, for the current query parameters.
+ * - **`getStatus()`**: Returns expanded status information for the current query parameters.
+ * - **`isDataExpired(override?)`**: Checks if the current data has expired based on `cacheTime`.
+ * - **`isStale(override?)`**: Checks if the current data is stale based on `staleTime`.
+ * - **`reset()`**: Resets the store to its initial state, clearing data and errors.
+ */
+export interface QueryStore<
+  TData,
+  TParams extends Record<string, unknown>,
+  S extends Omit<StoreState<TData, TParams>, keyof PrivateStoreState>,
+> extends UseBoundStore<StoreApi<S>> {
+  /**
+   * Indicates whether the store should actively fetch data.
+   * When `false`, the store won't automatically refetch data.
+   */
+  enabled: boolean;
+  /**
+   * Initiates a data fetch for the given parameters. If no parameters are provided, the store's
+   * current parameters are used.
+   * @param params - Optional parameters to pass to the fetcher function.
+   * @param options - Optional {@link FetchOptions} to customize the fetch behavior.
+   * @returns A promise that resolves when the fetch operation completes.
+   */
+  fetch: (params?: TParams, options?: FetchOptions) => Promise<void>;
+  /**
+   * Returns the cached data, if available, for the current query params.
+   * @returns The cached data, or `null` if no data is available.
+   */
+  getData: (params?: TParams) => TData | null;
+  /**
+   * Returns expanded status information for the currently specified query parameters. The raw
+   * status can be obtained by directly reading the `status` property.
+   * @example
+   * ```ts
+   * const isInitialLoad = useQueryStore(state => state.getStatus().isInitialLoad);
+   * ```
+   * @returns An object containing boolean flags for each status.
```

Comment:
`isInitialLoad` in particular or the whole status function?

Not sure I’ll keep either exactly as is currently. I think it’s all a bit redundant given `status` and `error` are exposed, and `isInitialLoad` in most cases is essentially just the absence of data. It’s mainly for consistency with react-query — personally I’d rarely use these helpers.

On `isInitialLoad`, you might want to for instance show a loading skeleton the first time the data loads (when there is no cached data), but not during refetches when cached data exists and can be displayed instead of a loading skeleton while new data is fetched.